src/background_script/index.ts

```diff
@@ -51,15 +52,20 @@ function openPrompt(request: PromptRequest): Promise<any> {
   });
 }
 
+// Need to store the payment request string in a background variable
+// to support the case where the user right-clicks on text without
+// selecting it first. In this situation, the background script does
+// not have access to the page dom to pull the string in the context
+// menu onclick handler.
+let currentPaymentRequest: string;
```

Comment:
I personally am not a fan of this line. I am not sure if global variables in background scripts are considered a code smell, but I tried searching for a better solution and couldn't come up with anything better. I'm very open to suggestions here if you have any.

Reply:
Nope, this makes sense to me. However, it might be good to refactor this section out into its own file. I've got another feature on another branch that does that so that background script isn't one big blob, I'll probably merge this as-is after I do some more testing and then refactor in a later commit. I appreciate you looking out for code quality though!

Reply:
Good point about the refactor. I agree 100%. I'll let you handle that and move on to another issue.

Thanks for the feedback.